spec/models/git_repository_spec.rb

```diff
@@ -188,5 +188,24 @@
       repo.refresh
       expect(repo.git_tags.collect(&:name)).to match_array(tag_list)
     end
+
+    context do
+      let(:dir_name) { repo.directory_name }
+
+      it "#destroy" do
+        allow(Dir).to receive(:exist?).with(dir_name).and_return(true)
+        expect(FileUtils).to receive(:rm_rf).with(dir_name)
+
+        repo.destroy
+      end
+
+      it "#destroy with failure" do
+        repo_id = repo.id
+        allow(repo).to receive(:delete_repo_dir).and_raise(MiqException::Error, "wham")
```

Comment:
@gmcculloug Isn't this a stub should a stub have an expect since we are defining the stub behavior here